src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -273,10 +254,14 @@ FindShardInterval(Datum partitionColumnValue, ShardInterval **shardIntervalCache
  * fire this.
  */
 static int
-FindShardIntervalIndex(Datum searchedValue, ShardInterval **shardIntervalCache,
-					   int shardCount, char partitionMethod, FmgrInfo *compareFunction,
-					   bool useBinarySearch)
+FindShardIntervalIndex(Datum searchedValue, DistTableCacheEntry *cacheEntry)
 {
+	ShardInterval **shardIntervalCache = cacheEntry->sortedShardIntervalArray;
+	int shardCount = cacheEntry->shardIntervalArrayLength;
+	char partitionMethod = cacheEntry->partitionMethod;
+	FmgrInfo *compareFunction = cacheEntry->shardIntervalCompareFunction;
+	bool useBinarySearch = (partitionMethod != DISTRIBUTE_BY_HASH ||
+							!cacheEntry->hasUniformHashDistribution);
```

Comment:
I think this change then essentially makes FastShardPruning obsolete, no?  I think we should just remove it in that case?